src/SequenceManager.cc

```diff
@@ -327,16 +348,34 @@ void SequenceManager::rename_sequence(UniqueId unique_id, const SequenceName& ne
         throw Error(gul14::cat("Cannot rename folder ", old_path.string(),
             " to ", new_path.string(), ": ", error.message()));
     }
+
+    // commit to local repository
+    auto commit_msg = stage_files_in_directory(old_seq_on_disk.path, "");
+    commit_msg += stage_files_in_directory(make_sequence_filename(new_name, unique_id), "");
+    if (commit_msg != "")
+        gl_.commit(gul14::cat("Rename ", old_seq_on_disk.name.string(), " to ", new_name.string(), ":", commit_msg));
 }
 
 void SequenceManager::rename_sequence(Sequence& sequence, const SequenceName& new_name)
-    const
 {
     rename_sequence(sequence.get_unique_id(), new_name);
     sequence.set_name(new_name);
 }
 
-void SequenceManager::store_sequence(const Sequence& seq) const
+void SequenceManager::store_sequence(const Sequence& seq)
+{
+    _store_sequence(seq);
+
+    // detect what has changed in the sequence
+    const auto dir_name = make_sequence_filename(seq.get_name(), seq.get_unique_id());
+    const auto commit_msg = stage_files_in_directory(dir_name, "");
```

Comment:
I stated this in a previous commit. I did not want to remove const qualifier from list_sequences because it  should be a const function from name and functionality. But it is using store_sequence to fix older sequence directories which do not have an ID yet

Reply:
And yes, we should discuss it. I am also not happy with it, it feels not finalised

Reply:
There is a pattern where a call to `store_sequence_impl(seq)` is followed by something like `auto dir_name = make_sequence_filename(seq.get_name(), seq.get_unique_id())`. Maybe all we need is to give the function a better name and to let it return the filename, e.g.:
```cpp
path write_sequence_folder(const Sequence&)
```
I am not really content with that name either and I don't know if it should rather return a path or a string... but the general direction feels right.

Reply:
As Lars pointed out the old-compat-stuff can go now, so there should not be an exception anymore?